### DISCUSSION TITLE
Connection to a network equipment through a middle server using a specfic pattern

### DISCUSSION THREAD

**Discussion by skadomers** [scrapli/scrapligo#93] [Q&A] [answered]
Hi,

Firstly, I'm pretty new in Go (and on github too) and I would like to adapt some of my python scripts using scrapli to Go.

In python, I used scrapli to execute command on networks equipments through a "middle server" overriding the "open_cmd" transport arguments with my connection pattern (ssh <my_user>@<middle_server> -t -- <equipment_name>)

I would like to do the same kind of connection with scrapligo, and I think I can use the "ExtraArgs" field of transport.System struct, but I don't know exactly how to do? Do you have any idea and is it the good way of using "ExtraArgs"?

Thanks for your help

Hervé

**Comment by carlmontanari** [comment, marked as answer]
Hey @skadomers 

Probably I would recommend just setting proxy jump things in your ssh config file, but if you want to do it this way, yep you are correct, you can override the `OpenBin` and `OpenArgs` of the system transport [here](https://github.com/scrapli/scrapligo/blob/main/transport/system.go#L34-L35).

You can also check out how it is done in kne [here](https://github.com/hellt/kne/blob/60cd2dc36d9922f2e01ef472612edf83c1c39cea/topo/node/node.go#L534-L550) - by passing options to the driver.

HTH!

**Reply by skadomers** [reply]
Hi @carlmontanari 

Thanks for your answer, I will try to override the OpenBin and OpenArgs arguments. Even with python I didn't find a way to work with the ssh config file and the proxy I have to go through.

Thanks so much for the help

Hervé
